Mathlib/NumberTheory/NumberField/AdeleRing.lean

```diff
@@ -0,0 +1,151 @@
+instance locallyCompactSpace [NumberField K] : LocallyCompactSpace (InfiniteAdeleRing K) :=
+  Pi.locallyCompactSpace_of_finite
+
+/-- The ring isomorphism between the infinite adele ring of a number field and the
+space `ℝ ^ r₁ × ℂ ^ r₂`, where `(r₁, r₂)` is the signature of the number field. -/
+abbrev ringEquiv_mixedSpace :
+    InfiniteAdeleRing K ≃+* mixedEmbedding.mixedSpace K :=
+  RingEquiv.trans
+    (RingEquiv.piEquivPiSubtypeProd (fun (v : InfinitePlace K) => IsReal v)
+      (fun (v : InfinitePlace K) => v.completion))
+    (RingEquiv.prodCongr
+      (RingEquiv.piCongrRight (fun ⟨_, hv⟩ => Completion.ringEquiv_real_of_isReal hv))
+      (RingEquiv.trans
+        (RingEquiv.piCongrRight (fun v => Completion.ringEquiv_complex_of_isComplex
+          ((not_isReal_iff_isComplex.1 v.2))))
+        (RingEquiv.piCongrLeft (fun _ => ℂ) <|
+          Equiv.subtypeEquivRight (fun _ => not_isReal_iff_isComplex))))
+
+@[simp]
+theorem ringEquiv_mixedSpace_apply (x : InfiniteAdeleRing K) :
+    ringEquiv_mixedSpace K x =
+      (fun (v : {w : InfinitePlace K // IsReal w}) => extensionEmbedding_of_isReal v.2 (x v),
+       fun (v : {w : InfinitePlace K // IsComplex w}) => extensionEmbedding v.1 (x v)) := rfl
+
+/-- Transfers the embedding of `x ↦ (x)ᵥ` of the number field `K` into its infinite adele
+ring to the mixed embedding `x ↦ (φᵢ(x))ᵢ` of `K` into the space `ℝ ^ r₁ × ℂ ^ r₂`, where
+`(r₁, r₂)` is the signature of `K` and `φᵢ` are the complex embeddings of `K`. -/
+theorem mixedEmbedding_eq_algebraMap_comp {x : K} :
+    mixedEmbedding K x = ringEquiv_mixedSpace K (algebraMap K _ x) := by
+  ext v <;> simp only [ringEquiv_mixedSpace_apply, algebraMap_apply,
+    ringEquiv_real_of_isReal, ringEquiv_complex_of_isComplex, extensionEmbedding,
+    extensionEmbedding_of_isReal, extensionEmbedding_of_comp, RingEquiv.coe_ofBijective,
+    RingHom.coe_mk, MonoidHom.coe_mk, OneHom.coe_mk, UniformSpace.Completion.extensionHom]
+  · rw [UniformSpace.Completion.extension_coe
+      (WithAbs.isUniformInducing_of_comp <| v.1.norm_embedding_of_isReal v.2).uniformContinuous x]
+    exact mixedEmbedding.mixedEmbedding_apply_ofIsReal _ _ _
+  · rw [UniformSpace.Completion.extension_coe
+      (WithAbs.isUniformInducing_of_comp <| v.1.norm_embedding_eq).uniformContinuous x]
+    exact mixedEmbedding.mixedEmbedding_apply_ofIsComplex _ _ _
+
+end InfiniteAdeleRing
+
+variable [NumberField K]
+
+/-! ## The adele ring  -/
+
+/-- The adele ring of a number field. -/
+def AdeleRing := InfiniteAdeleRing K × FiniteAdeleRing (𝓞 K) K
+
+namespace AdeleRing
+
+instance : CommRing (AdeleRing K) := Prod.instCommRing
+
+instance : Inhabited (AdeleRing K) := ⟨0⟩
+
+instance : TopologicalSpace (AdeleRing K) := instTopologicalSpaceProd
+
+instance : TopologicalRing (AdeleRing K) := instTopologicalRingProd
+
+instance : Algebra K (AdeleRing K) := Prod.algebra _ _ _
+
+@[simp]
+theorem algebraMap_fst_apply (x : K) : (algebraMap K (AdeleRing K) x).1 v = x := rfl
```

Comment:
```suggestion
theorem algebraMap_fst_apply (x : K) (v : InfinitePlace K) : 
    (algebraMap K (AdeleRing K) x).1 v = x := rfl
```
Furthermore, if you define v here it is consistent with the following definition of `algebraMap_snd_apply` since otherwise `v` comes before `x`.